expo/expo
# [web][camera] Fixed incorrect isIphoneX check and surrounded applyConstraints with try catch

# Why

To allow the camera to be loaded in the native-components-list for development purposes.

# How

Swapped out the isIphoneX check library [react-native-is-iphonex](https://www.npmjs.com/package/react-native-is-iphonex) with another more maintained library [react-native-iphone-x-helper](https://www.npmjs.com/package/react-native-iphone-x-helper) that was used by other dependencies in the project. Included small change wrapping `track.applyConstraints(...)` in a try catch with a warning in dev if `DOMException: setOptions failed`.

# Test Plan

* Start NCL `cd apps/native-component-list && yarn web`
* Open `Components`
* `Camera` option should not be greyed out and should now open
